### ISSUE TITLE
RepositoryManager temporarily disabled!

### ISSUE THREAD

**Issue by NOFUTURELLC** [Vector35/binaryninja-api#1083]
Binary Ninja Version: 1.1.1259 Personal, f51163e8
Platform: macOS High Sierra (10.13)

After the program was restarted, the previously installed plugins disappeared.
When I trying access RepositoryManager via console, get this exception

Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Volumes/BinaryNinja/Binary Ninja.app/Contents/MacOS/plugins/../../Resources/python/binaryninja/pluginmanager.py", line 204, in __init__
    raise Exception("RepositoryManager temporarily disabled!")
Exception: RepositoryManager temporarily disabled!

Next I try to delete .APP package and clear ~/L/ApplicationSuppport/
After reinstalling still get an exception.

**Comment by joshwatson** [comment]
They disabled it due to some library conflicts. It’s a known issue (hence the message)

**Comment by alilkenull** [comment]
Find build 1.1.1142 in my Downloads folder. Everything works!

In new version Exception are hardcoded into pluginmanager.py

	from binaryninja.enums import PluginType, PluginUpdateStatus
	def __init__(self, handle):
		raise Exception("RepoPlugin temporarily disabled!") # <--- !!!!
		self.handle = core.handle_of_type(handle, core.BNRepoPlugin)

When i comment this line, app are crashing when requesting for plugin updates.
If the problem is only with me, is it possible that this is some kind of limitation of my license. I would like to get a comment from developers.

**Comment by negasora** [comment]
It's been temporarily disabled due to some issues

**Comment by alilkenull** [comment]
Anyone know how match time it takes (resolving library conflicts). Plugins is very useful part of product for me, in some task make it stronger than, for ex. IDA. Because of this i became happy owner of license. But now I'll lost some functionality very important for me. I would be happy if a little to clarify the situation, the problem of how critical and how soon all Fixed. Thanks in advice. 
'''Sorry for the terrible English ;-)'''

**Comment by joshwatson** [comment]
You can manually install plugins still

**Comment by alilkenull** [comment]
No evety time plugin cannot load because:
```
Traceback (most recent call last):
  File "/Users/alikenull/Library/Application Support/Binary Ninja/plugins/binaryninja-bookmarks/__init__.py", line 120, in <module>
    bn.PluginCommand.register_for_address(
AttributeError: 'module' object has no attribute 'PluginCommand'
Python plugin 'binaryninja-bookmarks' could not be loaded
Traceback (most recent call last):
  File "/Users/alikenull/Library/Application Support/Binary Ninja/plugins/retdec_offline/__init__.py", line 12, in <module>
    from binaryninja.plugin import PluginCommand
ImportError: cannot import name PluginCommand
Python plugin 'retdec_offline' could not be loaded
Traceback (most recent call last):
  File "/Users/alikenull/Library/Application Support/Binary Ninja/plugins/linsweep/__init__.py", line 148, in <module>
    PluginCommand.register("Simple Linear Sweep - Cautious", "Search for existing prologues in text section", sweep_cat)
NameError: name 'PluginCommand' is not defined
Python plugin 'linsweep' could not be loaded
Traceback (most recent call last):
  File "/Users/alikenull/Library/Application Support/Binary Ninja/plugins/binaryninja-search/__init__.py", line 99, in <module>
    PluginCommand.register(name, description, do_stuff)
NameError: name 'PluginCommand' is not defined
Python plugin 'binaryninja-search' could not be loaded
Traceback (most recent call last):
  File "/Users/alikenull/Library/Application Support/Binary Ninja/plugins/bnil-graph/__init__.py", line 133, in <module>
    PluginCommand.register_for_address("BNIL Instruction Graph", "View BNIL Instruction Information", graph_bnil)
NameError: name 'PluginCommand' is not defined
Python plugin 'bnil-graph' could not be loaded
Traceback (most recent call last):
  File "/Users/alikenull/Library/Application Support/Binary Ninja/plugins/function-finder/__init__.py", line 1, in <module>
    from binaryninja.plugin import PluginCommand
ImportError: cannot import name PluginCommand
Python plugin 'function-finder' could not be loaded
Traceback (most recent call last):
  File "/Users/alikenull/Library/Application Support/Binary Ninja/plugins/WinAPI-Annotator/__init__.py", line 3, in <module>
    from binaryninja import PluginCommand
ImportError: cannot import name PluginCommand
Python plugin 'WinAPI-Annotator' could not be loaded

Like this...Stuck again. Patching sources my hands, not good idea, as we can se above.
Any solution. Please.

Great thanks to man who finally solve this task!

**Comment by alilkenull** [comment]
Sometimes uglypatching helps :-) Replace 2 files: plugin.py pluginmanager.py from 1.1.1142 to stable. Than, as some man advice in thread near, about new api dont remember exactly, replace all occurs of 'startup._init_plugins' to '_init_plugins'.

Put little more random plugins to user folder, bingo!
`Unable to find plugin binja-avr
Unable to find plugin liil
Loaded Python plugin 'binaryninja-bookmarks'
Loaded Python plugin 'retdec_offline'
Loaded Python plugin 'linsweep'
Loaded Python plugin 'binaryninja-search'
Loaded Python plugin 'bnil-graph'
Loaded Python plugin 'function-finder'
Loaded Python plugin 'WinAPI-Annotator'
`
I hope this information help peoples, also who bought license.

**Comment by alilkenull** [comment]
Here patched version: https://pastebin.com/jFWpckLL
Hope it help, who don't have old builds.


// One little file of thousands. I think no problem with leggaly of this upload ;-) Only for the best of reasons.

**Comment by joshwatson** [comment]
If you have a python module installed and you want to use it in Binary Ninja, just go to advanced settings in the Edit menu and change your Python interpreter to the one that has the module installed

**Comment by psifertex** [comment]
It's definitely coming back (just tagged this as 1.2) though it won't necessarily solve dependency problems for you yet. We might automate the "pip" section of the pre-requisites, but there are others that we likely won't be able to handle. That said, it should still make installation a lot easier. This is a blocker for #305.

**Comment by plafosse** [comment]
Fixed in v1.1.1761
